graphql/resolver.go

```diff
@@ -170,6 +171,40 @@ func New(apiURL string) Config {
 		}
 		return nil, Forbidden.Send(ctx, fmt.Sprintf("user %s does not have permission to access settings for the project %s", user.Username(), projectId))
 	}
+	c.Directives.RequireProjectAccessNew = func(ctx context.Context, obj interface{}, next graphql.Resolver, permission ProjectPermission, access AccessLevel) (interface{}, error) {
+		user := mustHaveUser(ctx)
+
+		args, isMap := obj.(map[string]interface{})
+		if !isMap {
+			return nil, InternalServerError.Send(ctx, "converting args into map")
+		}
+
+		requiredPermission, requiredLevel, err := getProjectPermissionLevel(permission, access)
+		if err != nil {
+			return nil, InputValidationError.Send(ctx, fmt.Sprintf("invalid permission and access level configuration: %s", err.Error()))
+		}
+
+		paramsMap, err := data.BuildProjectParameterMap(args)
+		if err != nil {
+			return nil, InputValidationError.Send(ctx, err.Error())
+		}
+
+		projectId, statusCode, err := data.GetProjectIdFromParams(ctx, paramsMap)
+		if err != nil {
+			return nil, mapHTTPStatusToGqlError(ctx, statusCode, err)
+		}
+
+		opts := gimlet.PermissionOpts{
+			Resource:      projectId,
+			ResourceType:  evergreen.ProjectResourceType,
+			Permission:    requiredPermission,
+			RequiredLevel: requiredLevel,
+		}
+		if user.HasPermission(opts) {
+			return next(ctx)
+		}
+		return nil, Forbidden.Send(ctx, fmt.Sprintf("user '%s' does not have permission to access %s for the project '%s'", user.Username(), strings.ToLower(permission.String()), projectId))
```

Comment:
nit: can we also include the permission string in '%s' ? I think even if this has a human readable name, it's helpful to know its a hardcoded value, in case it doesn't make sense in the sentence structure haha

Reply:
( think i didn't word this well so just to re-word, i meant using '%s' instead of %s)

Reply:
dw i got it hahaha!